controllers/whoToFollow/whoToFollowV2.js

```diff
@@ -0,0 +1,267 @@
+                            a.user_id,
+                            a.country_code,
+                            a.username,
+                            a.real_name,
+                            a.created_at,
+                            a.updated_at,
+                            a.last_active_at,
+                            a.status,
+                            a.profile_pic_path,
+                            a.profile_pic_asset_id,
+                            a.profile_pic_public_id,
+                            a.bio,
+                            a.is_banned,
+                            a.is_anonymous,
+                            a.encrypted,
+                            a.allow_anon_dm,
+                            a.only_received_dm_from_user_following,
+                            a.is_backdoor_user,
+                            a.followers_count,
+                            a.karma_score,
+                            b.topic_id
+                          FROM users a
+                          INNER JOIN user_topics b ON a.user_id = b.user_id
+                          WHERE 
+                            a.is_anonymous = false
+                            AND a.is_banned = false
+                            AND a.blocked_by_admin = false
+                            AND a.karma_score > 30
+                            AND a.user_id != :admin_user_id
+                            AND b.topic_id = :topic_id
+                          ORDER BY 
+                            a.last_active_at DESC,
+                            CASE WHEN a.profile_pic_path != '%default-profile-picture%' THEN 0 ELSE 1 END,
+                            a.followers_count DESC
+                          LIMIT :limit OFFSET :offset`;
+
+        const userTopicFollowerQueryResult = await sequelize.query(topicsQuery, {
+          type: sequelize.QueryTypes.SELECT,
+          replacements: {
+            topic_id: topic.topic_id,
+            admin_user_id: process.env.BETTER_ADMIN_ID,
+            limit,
+            offset
+          }
+        });
+        const userTopicFollower = userTopicFollowerQueryResult;
+
+        const tempUsers = [];
+        for (const indexUserTopic in userTopicFollower) {
+          const userTopic = userTopicFollower[indexUserTopic];
+
+          tempUsers.push({
+            ...userTopic,
+            viewtype: 'user'
+          });
+        }
+
+        if (tempUsers.length > 0) {
+          result.push({
+            viewtype: 'labeltopic',
+            name: topic.name,
+            id: topic.topic_id,
+            users_count: tempUsers.length,
+            users: tempUsers
+          });
+        }
+      }
+    }
+    // END TOPICS
+
+    //LOCATIONS
+    if (locations) {
+      const locationsDetail = `SELECT * FROM location WHERE location_id IN (:locations)`;
+
+      const locationsDetailQueryResult = await await sequelize.query(locationsDetail, {
+        type: sequelize.QueryTypes.SELECT,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous comment, there's a redundant `await` keyword here.
```diff
- const locationsDetailQueryResult = await await sequelize.query(locationsDetail, {
+ const locationsDetailQueryResult = await sequelize.query(locationsDetail, {
```

69,155:
Hardcoding the admin user ID from the environment variable directly into the SQL query might not be the best approach. It would be better to check if the user is an admin in your application logic rather than in the SQL query itself.



<!-- This is an auto-generated comment by OSS CodeRabbit -->